File: grpc/src/main/scala/grpc/lib/util/Logger.scala
package grpc
package lib
package util

object Logger {
  type LogLevel = LogLevel.Value
  object LogLevel extends Enumeration {
    final val ERROR = Value
    final val DEBUG = Value
  }

  private var maxLevel: LogLevel = LogLevel.ERROR

  def setMaxLevel(level: LogLevel) {
    maxLevel = level
  }

  def log(msg: String, level: LogLevel): Unit = {
    if (level <= maxLevel) {
      println(s"$level: $msg")
    }
  }
}

File: grpc/src/main/kotlin/grpc/lib/util/Logger.kt
package grpc.lib.util

object Logger {
    private var _maxLevel = LogLevel.ERROR

    fun setMaxLevel(level: LogLevel) {
        _maxLevel = level
    }

    fun log(msg: String?, level: LogLevel) {
        if (level <= _maxLevel) {
            when (level) {
                LogLevel.ERROR -> System.err.println(msg)
                else -> println(msg)
            }
        }
    }
}